frontend/src/metabase/containers/UndoListing.jsx

```diff
@@ -62,49 +54,57 @@ UndoToast.propTypes = {
 };
 
 function UndoToast({ undo, onUndo, onDismiss }) {
-  const style = undo.toastColor
-    ? { backgroundColor: color(undo.toastColor) }
-    : undefined;
   return (
-    <ToastCard dark data-testid="toast-undo" style={style}>
-      <CardContent>
-        <CardContentSide>
-          <CardIcon name={undo.icon || "check"} color="white" />
-          {renderMessage(undo)}
-        </CardContentSide>
-        <CardContentSide>
-          {undo.actions?.length > 0 && (
-            <UndoButton onClick={onUndo}>{t`Undo`}</UndoButton>
-          )}
-          <DismissIcon name="close" onClick={onDismiss} />
-        </CardContentSide>
-      </CardContent>
-    </ToastCard>
+    <Motion
+      defaultStyle={{ opacity: 0, translateY: 100 }}
+      style={{ opacity: spring(1), translateY: spring(0) }}
+    >
+      {({ translateY }) => (
+        <ToastCard
+          dark
+          data-testid="toast-undo"
+          style={{
+            ...{ transform: `translateY(${translateY}px)` },
+            ...(undo.toastColor
+              ? { backgroundColor: color(undo.toastColor) }
+              : {}),
+          }}
```

Comment:
I personally find that kind of conditional spreading pretty hard to parse. Could we refactor it somehow?

Reply:
Yeah I can add props to the styled component